### PR TITLE
add support to install xmlrpc with xml module

### PR DESCRIPTION
**Pull request by scambra**
Added to map.jinja so they can be installed with php:ng:modules
Should I add a state in php:ng for each extension?

### CODE REVIEW

**Comment by aboe76** [comment]
@scambra please do if you can.

**Comment by pather87** [comment]
Hi,

i just added my Pull request #177 for bz2 and dba support. So this pull request here would supersede it. 
I checked the map.jinja and saw that the lookup is only added for Debian, so this patch would not support Ubuntu / Redhat etc. Can you also add the support there ( check #177 for comparison )?

**Comment by scambra** [comment]
@pather87 I think your pull request is better, supports other distributions, probably better to add xmlrpc to your pull request. I see SUSE installs different packages for xml extension ['php5-xmlreader','php5-xmlwriter','php5-xmlrpc'], so maybe it's better to add 'php' + phpng_version + '-xmlrpc' to Debian and Ubuntu, I don't know if Gentoo or Redhat has separate package for xmlrpc

File: php/ng/bz2.sls [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{% set state = 'dba' %}
```
Comment (n-rodriguez) [inline php/ng/bz2.sls]: I think there's a typo here: it should be bz2

File: php/ng/dba.sls [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{% set state = 'adodb' %}
```
Comment (n-rodriguez) [inline php/ng/dba.sls]: I think there's a typo here: it should be dba

File: php/ng/bz2.sls (hunk 1 above)
Comment (scambra) [reply php/ng/bz2.sls]: You are right, I have fixed it. Although I think #177 is better, supports other distributions

File: php/ng/dba.sls (hunk 2 above)
Comment (scambra) [reply php/ng/dba.sls]: You are right, I have fixed it. Although I think #177 is better, supports other distributions

**Comment by myii** [comment]
@scambra Thanks for this and your comments regarding #177.  I've used your changes from this PR as mentioned in https://github.com/saltstack-formulas/php-formula/pull/177#issuecomment-506979472.  You'll see your contribution merged in this commit: 5e04187.

Now, you may want to just close this PR.  However, ideally you can continue from where we've got to with the merge of #177 and massage this PR into providing `xmlrpc` in a similar vein.  That would be really appreciated.  If you are interested in doing that, I can help extract the useful bits left over from the combination done in #177.

**Comment by scambra** [comment]
@myii how do you want to add xmlrpc? Add new line to pkgs for xmlrpc as I did? Or add xmlrpc as package dependency to xml line, as it's done for Suse?

Do you know how can I check where is php xmlrpc extension in Freebsd, Redhat and Gentoo? I know Arch has xmlrpc extension in php package

**Comment by myii** [comment]
@scambra Let's ask the main maintainers of this particular formula.
@aboe76 @alxwr How would you like to proceed in terms of the [previous comment](https://github.com/saltstack-formulas/php-formula/pull/175#issuecomment-507184031)?

**Comment by aboe76** [comment]
@scambra most of the time, look in to the distribution package repo is there is a package for the extension, ,or check when php is installed if the module is enabled/available...

php is a tricky in the sense that distribution have been packaging this in al sort of ways..

**Comment by scambra** [comment]
I have redo this PR to add xmlrpc package to xml module, as it's done in SUSE.

**Comment by n-rodriguez** [comment]
@myii I think it's mergeable, do you?

**Comment by myii** [comment]
Sorry @scambra, I didn't notice the new commit.  Thanks for the heads-up @n-rodriguez.

It's only failing the `commitlint`, so I'll fix that now when merging.  @scambra This repo uses `semantic-release` so we need to format our commit messages a certain way.  See https://github.com/saltstack-formulas/php-formula/blob/master/docs/CONTRIBUTING.rst#commit-message-formatting for more info.

So changing the commit message:

```diff
-add xmlrpc package for xml module, as it was done for SUSE
+feat(map): add xmlrpc package for xml module, as it was done for SUSE
```

**Comment by myii** [comment]
@n-rodriguez And now all of our warning messages from #185 need to be updated because we've got `v0.38.0` now... I think we'll wait until we get closer to the `php` deprecation first.

**Comment by n-rodriguez** [comment]
> @n-rodriguez And now all of our warning messages from #185 need to be updated because we've got `v0.38.0` now... I think we'll wait until we get closer to the `php` deprecation first.

can we extract the target version (0.38.x) in pillars?

**Comment by myii** [comment]
@n-rodriguez Good idea, we could put something in the map.  The only problem is that the `README` would still have to be updated manually.

**Comment by n-rodriguez** [comment]
> The only problem is that the `README` would still have to be updated manually.

sadly :/

**Pull request merged**